pkg/kubelet/apis/config/validation/validation.go

```diff
@@ -142,19 +142,40 @@ func ValidateKubeletConfiguration(kc *kubeletconfig.KubeletConfiguration) error
 	}
 
 	if localFeatureGate.Enabled(features.GracefulNodeShutdown) {
-		if kc.ShutdownGracePeriod.Duration < 0 || kc.ShutdownGracePeriodCriticalPods.Duration < 0 || kc.ShutdownGracePeriodCriticalPods.Duration > kc.ShutdownGracePeriod.Duration {
-			allErrors = append(allErrors, fmt.Errorf("invalid configuration: ShutdownGracePeriod %v must be >= 0, ShutdownGracePeriodCriticalPods %v must be >= 0, and ShutdownGracePeriodCriticalPods %v must be <= ShutdownGracePeriod %v", kc.ShutdownGracePeriod, kc.ShutdownGracePeriodCriticalPods, kc.ShutdownGracePeriodCriticalPods, kc.ShutdownGracePeriod))
+		if localFeatureGate.Enabled(features.PodPriorityBasedGracefulShutdown) {
+			if len(kc.PodPriorityShutdownGracePeriods) == 0 {
+				if kc.ShutdownGracePeriod.Duration < 0 || kc.ShutdownGracePeriodCriticalPods.Duration < 0 || kc.ShutdownGracePeriodCriticalPods.Duration > kc.ShutdownGracePeriod.Duration {
```

Comment:
is there a way to avoid duplicating these checks in both branches of this if statement. For example the check of 

```
kc.ShutdownGracePeriod.Duration > 0 && kc.ShutdownGracePeriod.Duration < time.Duration(time.Second)
```
and 

```
kc.ShutdownGracePeriod.Duration < 0 || kc.ShutdownGracePeriodCriticalPods.Duration < 0 || kc.ShutdownGracePeriodCriticalPods.Duration > kc.ShutdownGracePeriod.Duration
```

appear both hear and in the else clause.

Can we maybe refactor this in such a way to avoid the duplication? For example, first do check of 

```
if localFeatureGate.Enabled(features.GracefulNodeShutdown)
```

then have all the validation for `ShutdownGracePeriod` and `ShutdownGracePeriodCriticalPods` and then have a separate clause for

```
if localFeatureGate.Enabled(features.PodPriorityBasedGracefulShutdown)
```

which only checks the new fields (i.e. `PodPriorityShutdownGracePeriods`) and that both `PodPriorityShutdownGracePeriods` and old fields are not both set?

Reply:
I put myself around confused, thank you for helping me make it clear. 😂

Reply:
np, looks much better now :)